src/transition/kml_iss.cpp

```diff
@@ -0,0 +1,176 @@
+
+#include <kml_iss.h>
+
+#include <solver.h>
+
+
+static int sparse_err_t_2_int( const sparse_err_t err ) {
+
+	int result = 0;
+	switch ( err ) {
+		case NO_ERROR:
+		{
+			result = KMLSS_NO_ERROR;
+			break;
+		}
+		case NULL_ARGUMENT:
+		{
+			result = KMLSS_NULL_ARGUMENT;
+			break;
+		}
+		case ILLEGAL_ARGUMENT:
+		{
+			result = KMLSS_BAD_N;
+			break;
+		}
+		case OUT_OF_MEMORY:
+		{
+			result = KMLSS_NO_MEMORY;
+			break;
+		}
+		case UNKNOWN:
+		{
+			result = KMLSS_INTERNAL_ERROR;
+			break;
+		}
+	}
+	return result;
+}
+
+#define KML_CG_PREFIXED( name ) KmlIssCg ## name
+
+
+int KML_CG_PREFIXED(InitSI)( KmlSolverTask **handle, int n, const float *a, const int *ja, const int *ia ) {
+	sparse_err_t err = sparse_cg_init_sii( handle, n, a, ja, ia );
+	return sparse_err_t_2_int( err );
+}
+
+int KML_CG_PREFIXED(InitDI)( KmlSolverTask **handle, int n, const double *a, const int *ja, const int *ia ) {
+	sparse_err_t err = sparse_cg_init_dii( handle, n, a, ja, ia );
+	return sparse_err_t_2_int( err );
+}
+
+
+
+int KML_CG_PREFIXED(SetUserPreconditionerSI)(KmlSolverTask **, void *, int (*)(void *, float *)) {
+	return KMLSS_NOT_IMPLEMENTED;
+}
+
+int KML_CG_PREFIXED(SetUserPreconditionerDI)(KmlSolverTask **, void *, int (*)(void *, double *)) {
+	return KMLSS_NOT_IMPLEMENTED;
+}
+
+
+
+#define KMLSS_SET_ARG( nd, param, case_val, sparse_err_t_val )	\
+	if( nd != 1 ) { return KMLSS_BAD_DATA_SIZE; }				\
+	int err = KMLSS_NO_ERROR;									\
+	switch ( param )											\
+	{															\
+	case case_val:												\
+	{															\
+		const sparse_err_t r = sparse_err_t_val;				\
+		if( r != NO_ERROR ) {									\
+			err = ( r == NULL_ARGUMENT ) ?						\
+				KMLSS_NULL_ARGUMENT : KMLSS_INTERNAL_ERROR;		\
+		}														\
+		break;													\
+	}															\
+	default:													\
+		err = KMLSS_BAD_SELECTOR;								\
+	}															\
+	return err;
```

Comment:
The back-ticks are not space-aligned. Also, not better to define all macros somewhere at the top of this file?

Reply:
fixed